Vagrantfile

```diff
@@ -18,6 +18,34 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   config.vm.synced_folder ".", "/vagrant", disabled: true
   config.vm.synced_folder ".", "/srv/zulip"
 
+  proxy_config_file = ENV['HOME']+"/.zulip-vagrant-config"
+  http_proxy = https_proxy = no_proxy = ""
+
+  if File.file?(proxy_config_file)
+    IO.foreach(proxy_config_file) do |line|
+      line.chomp!
+      key, value = line.split(nil, 2)
+      case key
+      when /^([#;]|$)/; # ignore line
+      when "HTTP_PROXY"; http_proxy = value
+      when "HTTPS_PROXY"; https_proxy = value
+      when "NO_PROXY"; no_proxy = value
+      end
```

Comment:
YAML seems probably more finicky than your current implementation.  If Ruby supports something like https://docs.python.org/2/library/configparser.html, that could be good; otherwise I think we can try going with your implementation.